<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="arm">

    <ros2_control name="RobotSystem" type="system">

        <xacro:property name="PI" value="3.14159265359" />
        <xacro:if value="$(arg is_sim)">
            <hardware>
                <plugin>gazebo_ros2_control/GazeboSystem</plugin>
            </hardware>
        </xacro:if>
        <xacro:unless value="$(arg is_sim)">
            <hardware>
            <plugin>servo_controller/ServoInterface</plugin>
            <param name="port">/path/to/your/port</param>
            </hardware>
        </xacro:unless>

        <joint name="arm_joint_1">
            <command_interface name="position">
            <param name="min">-${PI / 2}</param>
            <param name="max">${PI / 2}</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>

        <joint name="arm_joint_2">
            <command_interface name="position">
            <param name="min">-${PI / 2}</param>
            <param name="max">${PI / 2}</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>

        <joint name="arm_joint_3">
            <command_interface name="position">
            <param name="min">-${PI / 2}</param>
            <param name="max">${PI / 2}</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>

        <joint name="arm_joint_4">
            <command_interface name="position">
            <param name="min">-${PI / 2}</param>
            <param name="max">0.0</param>
            </command_interface>
            <state_interface name="position"/>
        </joint>
    </ros2_control>

</robot>